pkg/work/spoke/controllers/manifestcontroller/manifestwork_controller.go

```diff
@@ -137,96 +136,45 @@ func (m *ManifestWorkController) sync(ctx context.Context, controllerContext fac
-		if errors.As(result.Error, &authError) {
-			klog.V(2).Infof("apply work %s fails with err: %v", manifestWorkName, result.Error)
-			result.Error = nil
-
-			if authError.RequeueTime < requeueTime {
-				requeueTime = authError.RequeueTime
+	var errs []error
+	for _, reconciler := range m.reconcilers {
+		manifestWork, newAppliedManifestWork, err = reconciler.reconcile(
+			ctx, controllerContext, manifestWork, newAppliedManifestWork)
+		var rqe commonhelper.RequeueError
+		if err != nil && errors.As(err, &rqe) {
+			if requeueTime > rqe.RequeueTime {
+				requeueTime = rqe.RequeueTime
 			}
+		} else if err != nil {
+			errs = append(errs, err)
 		}
-
-		// ignore server side apply conflict error since it cannot be resolved by error fallback.
-		var ssaConflict *apply.ServerSideApplyConflictError
-		if result.Error != nil && !errors.As(result.Error, &ssaConflict) {
-			errs = append(errs, result.Error)
-		}
-	}
-	manifestWork.Status.ResourceStatus.Manifests = helper.MergeManifestConditions(
-		manifestWork.Status.ResourceStatus.Manifests, newManifestConditions)
-	// handle condition type Applied
-	// #1: Applied - work status condition (with type Applied) is applied if all manifest conditions (with type Applied) are applied
-	if inCondition, exists := allInCondition(workapiv1.ManifestApplied, newManifestConditions); exists {
-		appliedCondition := metav1.Condition{
-			Type:               workapiv1.WorkApplied,
-			ObservedGeneration: manifestWork.Generation,
-			Status:             metav1.ConditionFalse,
-			Reason:             "AppliedManifestWorkFailed",
-			Message:            "Failed to apply manifest work",
-		}
-		if inCondition {
-			appliedCondition.Status = metav1.ConditionTrue
-			appliedCondition.Reason = "AppliedManifestWorkComplete"
-			appliedCondition.Message = "Apply manifest work complete"
-		}
-		meta.SetStatusCondition(&manifestWork.Status.Conditions, appliedCondition)
 	}
 
 	// Update work status
-	updated, err := m.manifestWorkPatcher.PatchStatus(ctx, manifestWork, manifestWork.Status, oldManifestWork.Status)
+	mwUpdated, err := m.manifestWorkPatcher.PatchStatus(ctx, manifestWork, manifestWork.Status, oldManifestWork.Status)
 	if err != nil {
-		errs = append(errs, fmt.Errorf("failed to update work status with err %w", err))
+		return err
 	}
 
-	if !updated && requeueTime < MaxRequeueDuration {
-		controllerContext.Queue().AddAfter(manifestWorkName, requeueTime)
+	amwUpdated, err := m.appliedManifestWorkPatcher.PatchStatus(
+		ctx, newAppliedManifestWork, newAppliedManifestWork.Status, appliedManifestWork.Status)
+	if err != nil {
+		return err
 	}
 
 	if len(errs) > 0 {
-		err = utilerrors.NewAggregate(errs)
-		klog.Errorf("Reconcile work %s fails with err: %v", manifestWorkName, err)
+		klog.Errorf("Reconcile work %s fails with err: %v", manifestWorkName, errs)
+		return utilerrors.NewAggregate(errs)
 	}
 
-	return err
+	if !mwUpdated && !amwUpdated && requeueTime < MaxRequeueDuration {
```

Comment:
In a case where a ManifestWork includes more than 1 manifest and one of them is failed to apply due to lack of permission, `wmUpdated` and `amwUpdated` will be `true` and `requeueTime` will be less than `MaxRequeueDuration`, but the requeue is still needed.

Reply:
I think so